dashboard-widgets.el

```diff
@@ -20,10 +20,13 @@
 
 ;;; Code:
 
+(require 'cl-lib)
+
 ;; Compiler pacifier
-(declare-function all-the-icons-dir-is-submodule "ext:all-the-icons.el")
+(declare-function all-the-icons-icon-for-dir "ext:all-the-icons.el")
 (declare-function all-the-icons-icon-for-file "ext:all-the-icons.el")
-(declare-function all-the-icons-match-to-alist "ext:all-the-icons.el")
+(declare-function all-the-icons-fileicon "ext:all-the-icons.el")
+(declare-function all-the-icons-octicon "ext:all-the-icons.el")
```

Comment:
Since you replace all the logic with `all-the-icons-icon-for-dir`, are those necessary? They are not being used anywhere now :thinking:

Reply:
No, `all-the-icons-fileicon` and `all-the-icons-octicon` are used in other functions.

Reply:
I see. I'm trying to debug why the linter is complaining :thinking: . So far, if those two lines are removed, it does not complain. But I don't see why it complains if they are being used later in the code :thinking: .

Reply:
I think I've found why it is failing. Those two functions are not really defined, they are generated using the macro `define-icon`. [Code here](https://github.com/domtronn/all-the-icons.el/blob/1416f37984486a44c6c0cbe0a2c985e82f965b6b/all-the-icons.el#L1019-L1024)

That is why the linter is not able to find the function declaration.